src/main/java/com/avioconsulting/mule/logger/internal/listeners/CustomLoggerAbstractNotificationListener.java

```diff
@@ -41,4 +50,66 @@ protected void logMessage(ComponentLocation location, Event event, String logMes
     logger.log(logProperties, messageAttributes, exceptionProperties, additionalProperties, config,
         location, correlationId);
   }
+
+  protected Map<String, String> getFlowLogAttributes(EnrichedServerNotification notification) {
+    Map<String, String> value = emptyAttributes;
+    String expression = config.getFlowLogAttributesMap().get(notification.getResourceIdentifier());
+    if (expression != null) {
+      TypedValue<Map<String, String>> evaluate = (TypedValue<Map<String, String>>) config.getExpressionManager()
+          .evaluate("#[" + expression + "]",
+              notification.getEvent().asBindingContext());
+      value = evaluate.getValue();
+      if (value == null)
+        value = emptyAttributes;
+    }
+    /**
+     * Flow name can contain wildcard (*)
+     * We only look for wildcard either starting of the string or ending of the
+     * string
+     * ex: mq-listener-* will look for all the flows that starts with mq-listener
+     * ex: *-mq-flow will look for all the flows that ends with -mq-flow
+     **/
+    else {
+      List<Map.Entry<String, String>> matchedEntries = config.getFlowLogAttributesMap().entrySet().stream()
+          .filter(entry -> matchWildcard(entry.getKey(), notification.getResourceIdentifier()))
+          .collect(Collectors.toList());
+      if (!matchedEntries.isEmpty()) {
+        expression = matchedEntries.get(0).getValue();
+        TypedValue<Map<String, String>> evaluate = (TypedValue<Map<String, String>>) config
+            .getExpressionManager()
+            .evaluate("#[" + expression + "]",
+                notification.getEvent().asBindingContext());
+        value = evaluate.getValue();
+        if (value == null)
+          value = emptyAttributes;
+      }
+    }
+    return value;
+  }
+
+  public boolean matchWildcard(String wildcardKey, String searchString) {
+    // Trim the wildcard key
+    String cleanWildcardKey = wildcardKey.trim();
+
+    // If wildcard key is just '*', match everything
+    if (cleanWildcardKey.equals("*")) {
+      return true;
+    }
```

Comment:
I think the wildcard `*` should be a catch-all if present, like the below order of evaluation - 
- exact match
- starts with
- ends with
- * as catch-all

Reply:
done, updated